From: Milian Wolff <mail@milianw.de>
Date: Mon, 22 Jun 2015 20:21:25 +0000
Subject: Remove unsafe function that returns reference to const.
X-Git-Url: http://quickgit.kde.org/?p=kdevplatform.git&a=commitdiff&h=e22645664c263884c7c3013fb07607304b6b0718
---
Remove unsafe function that returns reference to const.

This is undefined behavior, and newer compilers are emitting code
which leads to crashes then - and this is our fault, not theirs.

CCBUG: 347178
---


--- a/language/util/setrepository.h
+++ b/language/util/setrepository.h
@@ -50,10 +50,6 @@
     }
 
     T operator*() const {
-      return Conversion::toItem(*m_it);
-    }
-
-    const T& ref() const {
       return Conversion::toItem(*m_it);
     }
 